base/src/test/java/io/spine/base/EnvironmentTest.java

```diff
@@ -81,66 +76,70 @@ void setUp() {
-    }
+    @Nested
+    @DisplayName("tell that we are under tests if")
+    class UnderTests {
 
-    @Test
-    @DisplayName("tell that we are under tests if env. variable set to 1")
-    void environmentVar1() {
-        System.setProperty(ENV_KEY_TESTS, "1");
+        @Test
+        @DisplayName("the env. variable `ENV_TESTS_KEY` set directly")
+        void environmentVar() {
+            TESTS_VALUES.forEach(
+                    value -> {
+                        System.setProperty(TestsProperty.KEY, value);
+                        assertThat(environment.is(Tests.class)).isTrue();
+                    }
+            );
+        }
 
-        assertThat(environment.is(Tests.class)).isTrue();
-    }
+        @Test
+        @DisplayName("run under known testing framework")
+        void underTestFramework() {
+            // As we run this from under JUnit...
+            assertThat(environment.is(Tests.class)).isTrue();
+        }
 
-    @Test
-    @DisplayName("tell that we are under tests if run under known framework")
-    void underTestFramework() {
-        // As we run this from under JUnit...
-        assertThat(environment.is(Tests.class)).isTrue();
-    }
+        @Test
+        @DisplayName("when a deprecated `isTests()` method is used")
+        void underTestFrameworkDeprecated() {
+            @SuppressWarnings("deprecation")
+            boolean isTests = environment.isTests();
+            assertThat(isTests).isTrue();
+        }
 
-    @Test
-    @DisplayName("tell that we are not under tests if env set to something else")
-    void environmentVarUnknownValue() {
-        System.setProperty(ENV_KEY_TESTS, "neitherTrueNor1");
+        @Test
+        @DisplayName("if explicitly set to tests using the deprecated `setToTests()` method")
+        @SuppressWarnings("deprecation")
+        void explicitlySetTrue() {
+            environment.setToTests();
 
-        assertThat(environment.is(Production.class)).isTrue();
+            assertThat(environment.is(Tests.class)).isTrue();
+        }
     }
 
-    @Test
-    @DisplayName("tell that we are under tests when a deprecated method is used")
-    void underTestFrameworkDeprecated() {
-        @SuppressWarnings("deprecation")
-        boolean isTests = environment.isTests();
-        assertThat(isTests).isTrue();
-    }
+    @Nested
+    @DisplayName("tell that we are not under tests if")
+    class NotUnderTests {
 
-    @Test
-    @DisplayName("tell that we are under tests if explicitly set to tests using a deprecated method")
-    @SuppressWarnings("deprecation")
-    void explicitlySetTrue() {
-        environment.setToTests();
+        @Test
+        @DisplayName("the property value is set to an unsupported value")
+        void environmentVarUnknownValue() {
+            System.setProperty(TestsProperty.KEY, "neitherTrueNor1");
 
-        assertThat(environment.is(Tests.class)).isTrue();
-    }
+            assertThat(environment.is(Production.class)).isTrue();
```

Comment:
```suggestion
            assertThat(environment.is(Tests.class)).isFalse();
```

The same goes for all methods in the `NotUnderTests` class.